include/ck/tensor_operation/gpu/grid/gridwise_batched_mha_bwd_xdl_cshuffle_qloop_b2t_v1.hpp

```diff
@@ -1219,43 +1222,165 @@ struct GridwiseBatchedMultiheadAttentionBackward_Qloop_Xdl_CShuffle_V1
         return math::max(p_slash_sgrad_bytes_end, softmax_bytes_end, c_block_bytes_end);
     }
 
+    // D0
+    static constexpr auto D0M2 = Number<4>{};
+    static constexpr auto D0M1 = Number<MPerXdl>{} / D0M2;
+    static constexpr auto D0M0 = Number<MPerBlock>{} / Number<MPerXdl>{};
+
+    __host__ __device__ static constexpr auto
+    MakeD0GridDescriptor_M0_N0_M1_M2_N1_M3(const D0GridDesc_M_N& d0_grid_desc_m_n)
+    {
+        const auto M = d0_grid_desc_m_n.GetLength(I0);
+        const auto N = d0_grid_desc_m_n.GetLength(I1);
+
+        const auto MBlock = M / MPerBlock;
+        const auto NBlock = N / NPerBlock;
+
+        const auto d0_grid_desc_m0_n0_m1_m2_n1_m3 = transform_tensor_descriptor(
+            d0_grid_desc_m_n,
+            make_tuple(make_unmerge_transform(make_tuple(MBlock, D0M0, D0M1, D0M2)),
+                       make_unmerge_transform(make_tuple(NBlock, Number<NPerBlock>{}))),
+            make_tuple(Sequence<0>{}, Sequence<1>{}),
+            make_tuple(Sequence<0, 2, 3, 5>{}, Sequence<1, 4>{}));
+
+        return d0_grid_desc_m0_n0_m1_m2_n1_m3;
+    }
+
+    using D0GridDescriptor_M0_N0_M1_M2_N1_M3 =
+        remove_cvref_t<decltype(MakeD0GridDescriptor_M0_N0_M1_M2_N1_M3(D0GridDesc_M_N{}))>;
+
+    struct D0Loader
+    {
+        template <typename DataType>
+        struct TypeTransform
+        {
+            using Type = DataType;
+        };
+        template <>
+        struct TypeTransform<void>
+        {
+            using Type = ck::half_t;
+        };
+        static constexpr index_t NThreadClusterLengths = MPerXdl;
+        static_assert(MPerXdl <= KPerBlock);
+        static_assert(D0BlockTransferSrcScalarPerVector * NThreadClusterLengths <= NPerBlock,
+                      "D0BlockTransferSrcScalarPerVector * NThreadClusterLengths <= NPerBlock");
+        __host__ __device__ static constexpr auto GetD0BlockDescriptor_M0_N0_M1_M2_N1_M3()
```

Comment:
Rename to `GetD0BlockWriteDescriptor_M0_N0_M1_M2_N1_M3` to be easily comparable to `GetD0BlockReadDescriptor_N0_N1_M0_M1_M2`

Reply:
done